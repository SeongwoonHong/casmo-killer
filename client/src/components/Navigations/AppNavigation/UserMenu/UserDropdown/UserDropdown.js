/* eslint jsx-a11y/click-events-have-key-events: 0 */
/* eslint jsx-a11y/no-noninteractive-element-interactions: 0 */
/* eslint jsx-a11y/click-events-have-key-events: 0 */
/* eslint jsx-a11y/no-noninteractive-element-interactions: 0 */
import React, { Component } from 'react';
import classnames from 'classnames';
import { NavLink } from 'react-router-dom';

import './UserDropdown.scss';

class UserDropdown extends Component {

  render() {

    const {
      active, isLoggedIn, currentRoute, onLogout
    } = this.props;

    if (isLoggedIn) {
      return (
        <ul
          className={ classnames('User-dropdown main-menu', {
            active
          }) }
          onClick={ () => {
            setTimeout(() => {
              this.props.toggleUserMenu(false);
            });
          } }>
          <li className="User-dropdown__items--notification">
            <NavLink
              className="User-dropdown__items__links"
              to="/user/notifications">
              <i className="User-dropdown__items__links__icons material-icons">
                notifications
              </i>
              <span className="User-dropdown__items__links__text">
                Notifications
              </span>
              <span className="notification-badge">N</span>
            </NavLink>
          </li>
          <li className="User-dropdown__items">
            <NavLink
              className="User-dropdown__items__links"
              to="/user/dashboard">
              <i className="User-dropdown__items__links__icons material-icons">
                dashboard
              </i>
              <span className="User-dropdown__items__links__text">
                My Board
              </span>
            </NavLink>
          </li>
          <li className="User-dropdown__items">
            <NavLink
              className="User-dropdown__items__links"
              to="/user/settings">
              <i className="User-dropdown__items__links__icons material-icons">
                settings
              </i>
              <span className="User-dropdown__items__links__text">
                My Account
              </span>
            </NavLink>
          </li>
          <li className="User-dropdown__items">
            <a
              className="User-dropdown__items__links"
              role="button"
              tabIndex={ 0 }
              onKeyDown={ () => {} }
              onClick={ () => onLogout() }>
              <i className="User-dropdown__items__links__icons material-icons">
                power_settings_new
              </i>
              <span className="User-dropdown__items__links__text">
                Log out
              </span>
            </a>
          </li>
        </ul>
      );
    }

    return (
      <ul
        className={ classnames('User-dropdown main-menu', {
          active
        }) }
        onClick={ () => {
          setTimeout(() => {
            this.props.toggleUserMenu(false);
          });
        } }>
        <li className="User-dropdown__items">
          <NavLink
            className="User-dropdown__items__links"
            to={{
              pathname: '/user/auth/login',
              state: {
                from: currentRoute
              }
            }}>
            <i className="User-dropdown__items__links__icons material-icons">
              person
            </i>
            <span className="User-dropdown__items__links__text">
              Log In
            </span>
          </NavLink>
        </li>
        <li className="User-dropdown__items">
          <NavLink
            className="User-dropdown__items__links"
            to={{
              pathname: '/user/auth/signup',
              state: {
                from: currentRoute
              }
            }}>
            <i className="User-dropdown__items__links__icons material-icons">
              person_add
            </i>
            <span className="User-dropdown__items__links__text">
              Sign Up
            </span>
          </NavLink>
        </li>
      </ul>
    );

  }

}

export default UserDropdown;
