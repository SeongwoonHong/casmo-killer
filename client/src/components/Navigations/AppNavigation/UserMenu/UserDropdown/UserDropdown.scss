.User-dropdown {
    @media screen and (min-width: 992px) {
        @include abs-elem($top-nav-size + 2px, 0, auto, auto);
        background-color: #F5F5F5;
        border-bottom: 1px solid $main-menu-border-color;
        border-left: 1px solid $main-menu-border-color;
        box-shadow: 0 2px 10px rgba(0, 0, 0, 0.2);
        /* TODO: refactor this so that it doesn't conflic with menucontainer style */
        display: none !important;
        width: $user-dropdown-width;
        &:before {
            @include rectanglify(0);
            @include abs-elem(-5px, 17px, auto, auto);
            content: '';
            display: block;
            border-left: 5px solid transparent;
            border-right: 5px solid transparent;
            border-bottom: 5px solid #F5F5F5;
        }
        > li {
            > a {
                @include flex-box(flex-start);
                width: 100%;
                > * {
                    color: palette(Grey, 800);
                }
                &.active {
                    background-color: #fff;
                    border-top: 1px solid $main-menu-border-color;
                    > .material-icons,
                    > span {
                        color: palette(Grey, 800);
                    }
                }
            }
        }
        &.active {
            /* TODO: refactor this so that it doesn't conflic with menucontainer style */
            display: block !important;
        }
    }
}
