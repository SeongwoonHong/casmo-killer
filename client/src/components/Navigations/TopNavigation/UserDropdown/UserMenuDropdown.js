import React from 'react';
import { Link } from 'react-router-dom';

import PlainBtn from 'sharedComponents/PlainBtn';

import './UserMenuDropdown.scss';

const UserMenuDropdown = (props) => {

  if (props.user.isLoggedIn) {
    return (
      <ul className="dropdown-content logged-in">
        <li className="notification">
          <Link to="/user/notifications">
            <i className="material-icons">notifications</i>
            <span>Notifications</span>
            <span className="notification-badge">N</span>
          </Link>
        </li>
        <li>
          <Link to="/">
            <i className="material-icons">dashboard</i>
            <span>My Board</span>
          </Link>
        </li>
        <li>
          <Link to="/user/settings">
            <i className="material-icons">settings</i>
            <span>My Account</span>
          </Link>
        </li>
        <li className="divider" />
        <li>
          <PlainBtn onClick={ props.logout }>
            <i className="material-icons">power_settings_new</i>
            <span>Log out</span>
          </PlainBtn>
        </li>
      </ul>
    );
  }

  return (
    <ul className="dropdown-content">
      <li>
        <PlainBtn onClick={ () => props.openAuthModal() }>
          <i className="material-icons">person</i>
          <span>Log In</span>
        </PlainBtn>
      </li>
      <li className="divider" />
      <li>
        <PlainBtn onClick={ () => props.openAuthModal('register') }>
          <i className="material-icons">person_add</i>
          <span>Sign Up</span>
        </PlainBtn>
      </li>
    </ul>
  );

};

export default UserMenuDropdown;
