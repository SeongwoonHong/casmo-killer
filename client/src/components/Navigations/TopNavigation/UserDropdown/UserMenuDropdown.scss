.User-menu-dropdown {
    display: block;
    opacity: 1;
    right: 0;
    top: $top-nav-size;
    > li {
        min-height: 0;
        line-height: normal;
        > a {
            @include flex-box(flex-start);
            color: palette(Teal, 900);
            font-size: 12px;
            padding: 10px 12px;
            > .material-icons {
                font-size: 18px;
                margin-right: 10px;
            }
            > .notification-badge {
                left: 4px;
                top: 4px;
            }
        }
        &:hover {
            background-color: palette(Teal, 50);
        }
    }
    @media screen and (min-width: 768px) {
        > .notification {
            display: none;
        }
    }
    @media screen and (min-width: 992px) {
        &.logged-in {
            width: 254px;
        }
    }
}
