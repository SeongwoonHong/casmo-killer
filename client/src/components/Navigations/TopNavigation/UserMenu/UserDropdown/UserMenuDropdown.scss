.dropdown-content {
    display: block;
    opacity: 1;
    right: 0;
    top: 50px;
    > li {
        > a {
            font-size: 14px;
            > .material-icons {
                font-size: 20px;
                margin-right: 10px;
            }
            > .notification-badge {
                left: 4px;
                top: 4px;
            }
        }
        &:hover {
            background-color: #b2dfdb;
        }
    }
    @media screen and (min-width: 992px) {
        width: 158px;
        > li.notification {
            display: none;
        }
    }
}
