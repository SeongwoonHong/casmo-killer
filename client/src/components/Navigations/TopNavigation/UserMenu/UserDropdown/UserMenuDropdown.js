import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';

import './UserMenuDropdown.scss';

class UserMenuDropdown extends Component {

  render() {

    return (
      <ul className="teal lighten-5 dropdown-content">
        <li className="notification">
          <NavLink
            to="/"
            className="teal-text text-darken-4">
            <i className="material-icons">notifications</i>
            <span>Notifications</span>
            <span className="notification-badge">
              N
            </span>
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/"
            className="teal-text text-darken-4">
            <i className="material-icons">dashboard</i>
            <span>My Board</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/"
            className="teal-text text-darken-4">
            <i className="material-icons">account_circle</i>
            <span>Profiles</span>
          </NavLink>
        </li>
        <li>
          <NavLink
            to="/"
            className="teal-text text-darken-4">
            <i className="material-icons">settings</i>
            <span>Settings</span>
          </NavLink>
        </li>
        <li className="divider" />
        <li>
          <a
            role="button"
            tabIndex={ 0 }
            onClick={ this.props.logout }
            onKeyDown={ () => {
            } }
            className="teal-text text-darken-4">
            <i className="material-icons">power_settings_new</i>
            <span>Log out</span>
          </a>
        </li>
      </ul>
    );

  }

}

export default UserMenuDropdown;
