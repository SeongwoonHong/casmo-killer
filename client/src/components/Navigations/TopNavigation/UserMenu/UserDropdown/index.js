import { connect } from 'react-redux';
import UserMenuDropdown from './UserMenuDropdown';

import * as actions from '../../../../../actions';

const mapStateToProps = (state) => {
  return {
    layout: state.layout
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    logout: () => dispatch(actions.logout())
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(UserMenuDropdown);
