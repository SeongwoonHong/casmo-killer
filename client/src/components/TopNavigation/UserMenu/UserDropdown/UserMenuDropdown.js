import React from 'react';
import { NavLink } from 'react-router-dom';

import './UserMenuDropdown.scss';

const NotificationBadge = (showNotification) => {
  if (showNotification) {
    return (
      <li>
        <NavLink
          to="/"
          className="teal-text text-darken-4">
          <i className="material-icons">notifications</i>
          <span>Notifications</span>
          <span className="notification-badge">
            N
          </span>
        </NavLink>
      </li>
    );
  }
  return null;
};

const UserMenuDropdown = ({ showNotification }) => {
  return (
    <ul className="teal lighten-5 dropdown-content">
      { NotificationBadge(showNotification) }
      <li>
        <NavLink
          to="/"
          className="teal-text text-darken-4">
          <i className="material-icons">dashboard</i>
          <span>My Board</span>
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/"
          className="teal-text text-darken-4">
          <i className="material-icons">account_circle</i>
          <span>Profiles</span>
        </NavLink>
      </li>
      <li>
        <NavLink
          to="/"
          className="teal-text text-darken-4">
          <i className="material-icons">settings</i>
          <span>Settings</span>
        </NavLink>
      </li>
      <li className="divider" />
      <li>
        <NavLink
          to="/"
          className="teal-text text-darken-4">
          <i className="material-icons">power_settings_new</i>
          <span>Log out</span>
        </NavLink>
      </li>
    </ul>
  );
};

export default UserMenuDropdown;
