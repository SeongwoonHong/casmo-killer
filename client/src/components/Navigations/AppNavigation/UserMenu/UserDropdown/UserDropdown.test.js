/* eslint react/jsx-boolean-value: 0 */
import React from 'react';
import { shallow } from 'enzyme';

import UserDropdown from './index';

describe('<UserDropdown />', () => {

  let component = null;

  beforeEach(() => {
    component = shallow(
      <UserDropdown
        active={ true }
        isLoggedIn={ true }
        currentRoute="/"
        onLogout={ () => {} } />
    );
  });

  it('renders the component consistently', () => {

    expect(component).toMatchSnapshot();

  });

});
