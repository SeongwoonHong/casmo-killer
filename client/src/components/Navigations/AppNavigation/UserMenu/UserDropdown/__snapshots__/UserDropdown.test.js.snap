// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserDropdown /> renders the component consistently 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UserDropdown
    active={true}
    currentRoute="/"
    isLoggedIn={true}
    onLogout={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="User-dropdown__items"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="User-dropdown__items__links"
                    to="/user/notifications"
          >
                    <i
                              className="User-dropdown__items__links__icons material-icons"
                    >
                              notifications
                    </i>
                    <span
                              className="User-dropdown__items__links__text"
                    >
                              Notifications
                    </span>
                    <span
                              className="notification-badge"
                    >
                              N
                    </span>
          </NavLink>
</li>,
        <li
          className="User-dropdown__items"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="User-dropdown__items__links"
                    to="/user/dashboard"
          >
                    <i
                              className="User-dropdown__items__links__icons material-icons"
                    >
                              dashboard
                    </i>
                    <span
                              className="User-dropdown__items__links__text"
                    >
                              My Board
                    </span>
          </NavLink>
</li>,
        <li
          className="User-dropdown__items"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="User-dropdown__items__links"
                    to="/user/settings"
          >
                    <i
                              className="User-dropdown__items__links__icons material-icons"
                    >
                              settings
                    </i>
                    <span
                              className="User-dropdown__items__links__text"
                    >
                              My Account
                    </span>
          </NavLink>
</li>,
        <li
          className="User-dropdown__items"
>
          <a
                    className="User-dropdown__items__links"
                    onClick={[Function]}
                    onKeyDown={[Function]}
                    role="button"
                    tabIndex={0}
          >
                    <i
                              className="User-dropdown__items__links__icons material-icons"
                    >
                              power_settings_new
                    </i>
                    <span
                              className="User-dropdown__items__links__text"
                    >
                              Log out
                    </span>
          </a>
</li>,
      ],
      "className": "User-dropdown main-menu active",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="User-dropdown__items__links"
            to="/user/notifications"
>
            <i
                        className="User-dropdown__items__links__icons material-icons"
            >
                        notifications
            </i>
            <span
                        className="User-dropdown__items__links__text"
            >
                        Notifications
            </span>
            <span
                        className="notification-badge"
            >
                        N
            </span>
</NavLink>,
          "className": "User-dropdown__items",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              <i
                className="User-dropdown__items__links__icons material-icons"
>
                notifications
</i>,
              <span
                className="User-dropdown__items__links__text"
>
                Notifications
</span>,
              <span
                className="notification-badge"
>
                N
</span>,
            ],
            "className": "User-dropdown__items__links",
            "to": "/user/notifications",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "notifications",
                "className": "User-dropdown__items__links__icons material-icons",
              },
              "ref": null,
              "rendered": "notifications",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Notifications",
                "className": "User-dropdown__items__links__text",
              },
              "ref": null,
              "rendered": "Notifications",
              "type": "span",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "N",
                "className": "notification-badge",
              },
              "ref": null,
              "rendered": "N",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="User-dropdown__items__links"
            to="/user/dashboard"
>
            <i
                        className="User-dropdown__items__links__icons material-icons"
            >
                        dashboard
            </i>
            <span
                        className="User-dropdown__items__links__text"
            >
                        My Board
            </span>
</NavLink>,
          "className": "User-dropdown__items",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              <i
                className="User-dropdown__items__links__icons material-icons"
>
                dashboard
</i>,
              <span
                className="User-dropdown__items__links__text"
>
                My Board
</span>,
            ],
            "className": "User-dropdown__items__links",
            "to": "/user/dashboard",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "dashboard",
                "className": "User-dropdown__items__links__icons material-icons",
              },
              "ref": null,
              "rendered": "dashboard",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Board",
                "className": "User-dropdown__items__links__text",
              },
              "ref": null,
              "rendered": "My Board",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="User-dropdown__items__links"
            to="/user/settings"
>
            <i
                        className="User-dropdown__items__links__icons material-icons"
            >
                        settings
            </i>
            <span
                        className="User-dropdown__items__links__text"
            >
                        My Account
            </span>
</NavLink>,
          "className": "User-dropdown__items",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "activeClassName": "active",
            "ariaCurrent": "true",
            "children": Array [
              <i
                className="User-dropdown__items__links__icons material-icons"
>
                settings
</i>,
              <span
                className="User-dropdown__items__links__text"
>
                My Account
</span>,
            ],
            "className": "User-dropdown__items__links",
            "to": "/user/settings",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "settings",
                "className": "User-dropdown__items__links__icons material-icons",
              },
              "ref": null,
              "rendered": "settings",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "My Account",
                "className": "User-dropdown__items__links__text",
              },
              "ref": null,
              "rendered": "My Account",
              "type": "span",
            },
          ],
          "type": [Function],
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            className="User-dropdown__items__links"
            onClick={[Function]}
            onKeyDown={[Function]}
            role="button"
            tabIndex={0}
>
            <i
                        className="User-dropdown__items__links__icons material-icons"
            >
                        power_settings_new
            </i>
            <span
                        className="User-dropdown__items__links__text"
            >
                        Log out
            </span>
</a>,
          "className": "User-dropdown__items",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <i
                className="User-dropdown__items__links__icons material-icons"
>
                power_settings_new
</i>,
              <span
                className="User-dropdown__items__links__text"
>
                Log out
</span>,
            ],
            "className": "User-dropdown__items__links",
            "onClick": [Function],
            "onKeyDown": [Function],
            "role": "button",
            "tabIndex": 0,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "power_settings_new",
                "className": "User-dropdown__items__links__icons material-icons",
              },
              "ref": null,
              "rendered": "power_settings_new",
              "type": "i",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Log out",
                "className": "User-dropdown__items__links__text",
              },
              "ref": null,
              "rendered": "Log out",
              "type": "span",
            },
          ],
          "type": "a",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="User-dropdown__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="User-dropdown__items__links"
                        to="/user/notifications"
            >
                        <i
                                    className="User-dropdown__items__links__icons material-icons"
                        >
                                    notifications
                        </i>
                        <span
                                    className="User-dropdown__items__links__text"
                        >
                                    Notifications
                        </span>
                        <span
                                    className="notification-badge"
                        >
                                    N
                        </span>
            </NavLink>
</li>,
          <li
            className="User-dropdown__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="User-dropdown__items__links"
                        to="/user/dashboard"
            >
                        <i
                                    className="User-dropdown__items__links__icons material-icons"
                        >
                                    dashboard
                        </i>
                        <span
                                    className="User-dropdown__items__links__text"
                        >
                                    My Board
                        </span>
            </NavLink>
</li>,
          <li
            className="User-dropdown__items"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="User-dropdown__items__links"
                        to="/user/settings"
            >
                        <i
                                    className="User-dropdown__items__links__icons material-icons"
                        >
                                    settings
                        </i>
                        <span
                                    className="User-dropdown__items__links__text"
                        >
                                    My Account
                        </span>
            </NavLink>
</li>,
          <li
            className="User-dropdown__items"
>
            <a
                        className="User-dropdown__items__links"
                        onClick={[Function]}
                        onKeyDown={[Function]}
                        role="button"
                        tabIndex={0}
            >
                        <i
                                    className="User-dropdown__items__links__icons material-icons"
                        >
                                    power_settings_new
                        </i>
                        <span
                                    className="User-dropdown__items__links__text"
                        >
                                    Log out
                        </span>
            </a>
</li>,
        ],
        "className": "User-dropdown main-menu active",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="User-dropdown__items__links"
              to="/user/notifications"
>
              <i
                            className="User-dropdown__items__links__icons material-icons"
              >
                            notifications
              </i>
              <span
                            className="User-dropdown__items__links__text"
              >
                            Notifications
              </span>
              <span
                            className="notification-badge"
              >
                            N
              </span>
</NavLink>,
            "className": "User-dropdown__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                <i
                  className="User-dropdown__items__links__icons material-icons"
>
                  notifications
</i>,
                <span
                  className="User-dropdown__items__links__text"
>
                  Notifications
</span>,
                <span
                  className="notification-badge"
>
                  N
</span>,
              ],
              "className": "User-dropdown__items__links",
              "to": "/user/notifications",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "notifications",
                  "className": "User-dropdown__items__links__icons material-icons",
                },
                "ref": null,
                "rendered": "notifications",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Notifications",
                  "className": "User-dropdown__items__links__text",
                },
                "ref": null,
                "rendered": "Notifications",
                "type": "span",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "N",
                  "className": "notification-badge",
                },
                "ref": null,
                "rendered": "N",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="User-dropdown__items__links"
              to="/user/dashboard"
>
              <i
                            className="User-dropdown__items__links__icons material-icons"
              >
                            dashboard
              </i>
              <span
                            className="User-dropdown__items__links__text"
              >
                            My Board
              </span>
</NavLink>,
            "className": "User-dropdown__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                <i
                  className="User-dropdown__items__links__icons material-icons"
>
                  dashboard
</i>,
                <span
                  className="User-dropdown__items__links__text"
>
                  My Board
</span>,
              ],
              "className": "User-dropdown__items__links",
              "to": "/user/dashboard",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "dashboard",
                  "className": "User-dropdown__items__links__icons material-icons",
                },
                "ref": null,
                "rendered": "dashboard",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "My Board",
                  "className": "User-dropdown__items__links__text",
                },
                "ref": null,
                "rendered": "My Board",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="User-dropdown__items__links"
              to="/user/settings"
>
              <i
                            className="User-dropdown__items__links__icons material-icons"
              >
                            settings
              </i>
              <span
                            className="User-dropdown__items__links__text"
              >
                            My Account
              </span>
</NavLink>,
            "className": "User-dropdown__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": Array [
                <i
                  className="User-dropdown__items__links__icons material-icons"
>
                  settings
</i>,
                <span
                  className="User-dropdown__items__links__text"
>
                  My Account
</span>,
              ],
              "className": "User-dropdown__items__links",
              "to": "/user/settings",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "settings",
                  "className": "User-dropdown__items__links__icons material-icons",
                },
                "ref": null,
                "rendered": "settings",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "My Account",
                  "className": "User-dropdown__items__links__text",
                },
                "ref": null,
                "rendered": "My Account",
                "type": "span",
              },
            ],
            "type": [Function],
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              className="User-dropdown__items__links"
              onClick={[Function]}
              onKeyDown={[Function]}
              role="button"
              tabIndex={0}
>
              <i
                            className="User-dropdown__items__links__icons material-icons"
              >
                            power_settings_new
              </i>
              <span
                            className="User-dropdown__items__links__text"
              >
                            Log out
              </span>
</a>,
            "className": "User-dropdown__items",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <i
                  className="User-dropdown__items__links__icons material-icons"
>
                  power_settings_new
</i>,
                <span
                  className="User-dropdown__items__links__text"
>
                  Log out
</span>,
              ],
              "className": "User-dropdown__items__links",
              "onClick": [Function],
              "onKeyDown": [Function],
              "role": "button",
              "tabIndex": 0,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "power_settings_new",
                  "className": "User-dropdown__items__links__icons material-icons",
                },
                "ref": null,
                "rendered": "power_settings_new",
                "type": "i",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Log out",
                  "className": "User-dropdown__items__links__text",
                },
                "ref": null,
                "rendered": "Log out",
                "type": "span",
              },
            ],
            "type": "a",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
